pkg/config/config_template.yaml

```diff
@@ -1,424 +1,781 @@
 {{ if .Common }}
-# The Datadog api key to associate your Agent's data with your organization.
-# Can be found here:
-# https://app.datadoghq.com/account/settings
+###############################
+## Agent Basic Configuration ##
+###############################
+
+## @param api_key - string - required
+## The Datadog api key to associate your Agent's data with your organization.
+## Can be found here: https://app.datadoghq.com/account/settings
+#
 api_key:
 
-# The site of the Datadog intake to send Agent data to.
-# Defaults to 'datadoghq.com', set to 'datadoghq.eu' to send data to the EU site.
-# site: datadoghq.com
+## @param site - string - optional - default: datadoghq.com
+## The site of the Datadog intake to send Agent data to.
+## Set to 'datadoghq.eu' to send data to the EU site.
+#
+#  site: datadoghq.com
+
+## @param dd_url - string - optional - default: https://app.datadoghq.com
+## The host of the Datadog intake server to send Agent data to, only set this option
+## if you need the Agent to send data to a custom URL, it overrides the site setting defined in "site".
+#
+#  dd_url: https://app.datadoghq.com
 
-# The host of the Datadog intake server to send Agent data to, only set this option
-# if you need the Agent to send data to a custom URL.
-# Overrides the site setting defined in "site".
-# dd_url: https://app.datadoghq.com
+## @param proxy - custom object - optional
```

Comment:
2c: I think it makes sense to put a space between the two hashes (#). It would align better with the actual setting, as it would in a normal yaml file.

Reply:
I think the general idea from a user standpoint is that you would be able to delete the first two characters on any line to enable any commented out setting. Without the added space I have suggested, this wouldn’t work in areas with comments because you’d be deleting the hash symbol too, creating invalid yaml. Does this make sense?